kotlin/workflow-runtime/src/jmh/java/com/squareup/workflow/WorkflowNodeBenchmark.kt

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow
+
+import com.squareup.workflow.FractalWorkflow.Props
+import com.squareup.workflow.FractalWorkflow.Props.DO_NOT_RENDER_LEAVES
+import com.squareup.workflow.FractalWorkflow.Props.RENDER_LEAVES
+import com.squareup.workflow.FractalWorkflow.Props.RUN_WORKERS
+import com.squareup.workflow.WorkflowAction.Companion.noAction
+import com.squareup.workflow.internal.WorkflowNode
+import com.squareup.workflow.internal.id
+import kotlinx.coroutines.Dispatchers.Unconfined
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.suspendCancellableCoroutine
+import org.openjdk.jmh.annotations.Benchmark
+import org.openjdk.jmh.annotations.BenchmarkMode
+import org.openjdk.jmh.annotations.Fork
+import org.openjdk.jmh.annotations.Measurement
+import org.openjdk.jmh.annotations.Mode.AverageTime
+import org.openjdk.jmh.annotations.OutputTimeUnit
+import org.openjdk.jmh.annotations.Param
+import org.openjdk.jmh.annotations.Scope
+import org.openjdk.jmh.annotations.Setup
+import org.openjdk.jmh.annotations.State
+import org.openjdk.jmh.annotations.Warmup
+import java.util.concurrent.TimeUnit.MICROSECONDS
+import java.util.concurrent.TimeUnit.SECONDS
+
+@Fork(value = 1)
+@State(Scope.Thread)
+@BenchmarkMode(AverageTime)
+@Warmup(iterations = 3, time = 5, timeUnit = SECONDS)
+@Measurement(iterations = 3, time = 5, timeUnit = SECONDS)
+@OutputTimeUnit(MICROSECONDS)
+open class WorkflowNodeBenchmark {
+
+  /**
+   * Used to parameterize the benchmarks.
+   *
+   * @param childCount The number of children _each workflow in the tree_ should have.
+   * @param depth The depth of the workflow tree.
+   */
+  enum class TreeShape(
+    val childCount: Int,
+    val depth: Int
+  ) {
+    DEEP(childCount = 1, depth = 100),
+    BUSHY(childCount = 100, depth = 1),
+    SQUARE(childCount = 5, depth = 5);
```

Comment:
Seems like it would be more consistent for SQUARE to be 10 x 10, so that you're always checking 100 nodes.

Reply:
And if that's a bad idea for some reason, be good to document why.

Reply:
That's 10^10=10,000,000,000 nodes, which takes 5 minutes and then runs out of memory :P 

But yea we should keep it to the same overall node count for each.

Reply:
Done.